src/Libraries/Microsoft.Extensions.AI.Evaluation.Reporting/CSharp/ScenarioRunResult.cs

```diff
@@ -81,6 +83,11 @@ public ScenarioRunResult(
     {
     }
 
+    /// <summary>
+    /// Gets or sets the version of the format used to persist the current <see cref="ScenarioRunResult"/>.
+    /// </summary>
+    public int? FormatVersion { get; set; } = formatVersion ?? Defaults.ReportingFormatVersion;
```

Comment:
Should FormatVersion be incorporated into the cache key? Otherwise, who's consuming this?

Reply:
> We are consuming this from previously persisted copies of our ScenarioRunResults (i.e. stored on disk or Azure storage, and created by potentially earlier versions of our library).

Sorry, I'm not following. Nothing appears to be using this property at all. Am I missing it? And if nothing's using it, why does the ctor parameter / property need to exist?

Reply:
It is for future proofing. JSON persistence writes the value of this property to disk. In future versions, we will check the value of the property and determine whether we need to modify the data before using it. 

We have run into this issue already with the ChatResponse change, and this is intended to help us solve the problem in the future.

Reply:
>  JSON persistence writes the value of this property to disk.

Where does that happen?

Reply:
Oh... is it that instances of this object are being JSON serialized, and so the property comes along for the ride? Got it.

Reply:
It happens any time a `ScenarioRunResult` is saved to an `IResultStore`. For example, [here](https://github.com/dotnet/extensions/blob/e351c2506f2af13be2645fbd554c89fb9bd32abf/src/Libraries/Microsoft.Extensions.AI.Evaluation.Reporting/CSharp/Storage/DiskBasedResultStore.cs#L101) in the DiskBased version.